rainier-core/src/main/scala/com/stripe/rainier/core/Model.scala

```diff
@@ -4,6 +4,9 @@ import com.stripe.rainier.compute._
 
 case class Model(targets: List[Target]) {
   def merge(other: Model) = Model(targets ++ other.targets)
+
+  def predict[T, U](value: T)(implicit tg: ToGenerator[T, U]): Prediction[U] =
```

Comment:
To make sure I udnerstand, if T is a generator, then it'll be possible to predict many values for whatever the generator emits?

Reply:
So,

`
`Model.observe(ys, dist).predict(g)` is equivalent to  something like

```
for {
 ...
 _ <- Events.observe(ys, dist)
} yield g
```

So if `g` is a `Generator[T]`, then we will produce a simulated/predicted `T` for each sample. Does that help?

Reply:
Yep, that's what I figured, thanks!